### PR TITLE
🐛 fix(v1.6): remediate comprehensive review findings

### PR DESCRIPTION
**Pull request by scttbnsn**
## Summary

- closes all repository-actionable findings from the comprehensive v1.6 review
- fixes runtime identity, lifecycle, cache, and state-transition correctness gaps
- hardens API/UI/demo contracts, error handling, security posture, and accessibility
- promotes the exact approved candidate in the release workflow and strengthens release evidence checks
- updates authentication, security, support, and acceptance documentation to match the shipped contracts
- includes the exact E2E expectation alignment from #573 (`460fb91a`), so this PR supersedes #573

Once merged, the existing `dev/v1.6` → `main` release PR #572 will automatically update to this remediated tree.

## Verification

- app: 12,287 tests across 392 files; 100% statements, branches, functions, and lines
- UI: 4,273 tests across 213 files; 100% statements, branches, functions, and lines
- workflow tests: 39 tests across 9 files
- root release/docs scripts: 107 tests
- demo behavioral/security tests: 5 tests; build green
- app and UI production builds green
- UI typecheck green
- actionlint green
- zizmor: no findings (offline mode; configured suppressions reported)
- repository pre-push policy gate green in full

## Release handling

This is the complete remediation path for v1.6. Do not merge #572 until this PR is integrated and its updated candidate has completed the required exact-SHA acceptance window.

## Changelog

- ✨ Added demo contract/security test scripts and made demo contract + production build first-class CI gates.
- ✨ Added release-candidate/GA promotion inputs and end-to-end release evidence checks for exact-SHA acceptance.
- ✨ Added OpenAPI update-gating models (`UpdateBlocker`, `UpdateEligibility`) and exposed `ContainerResource.tagPinGated` + `updateEligibility`.
- ✨ Added digest-cache poll-cycle “generation/handle” plumbing across registries to prevent cross-cycle invalidation.
- ✨ Added contract tests for release workflow wiring (demo build gates; release-cut retry naming; digest/tag propagation; recoverable finalization ordering).
- 🔧 Changed release-cut to “stage first, finalize last” with idempotent, digest-validated, recoverable GA finalization (candidate digest matching + exact image digest publishing + draft/public guards).
- 🔧 Changed preview error actions from `{ label, href }` to typed `{ code, href }` (API, OpenAPI schemas, UI normalization, i18n presentation).
- 🔧 Changed authentication/upgrades documentation to consistently describe fail-closed behavior (protected routes: `401`; `/health`: `503`; discovery/status remains public).
- 🔧 Changed SSE client lifecycle to track/route events only from the active stream (and ignore late events after stop).
- 🔧 Changed container candidate identity logic to ignore `suggestedTag`, gate `created` identity changes to legacy (no-digest) cases, and apply watcher-scoped identity keys.
- 🔧 Changed notification/template preview handling to prevent stale async results (generation invalidation) and updated MSW mocks to validate preview/persistence shapes.
- 🐛 Fixed overlapping digest-cache poll cycle end behavior and registry end-handle correctness.
- 🐛 Fixed preview race conditions across container/notification/template selection changes (stale results cannot overwrite newer state).
- 🐛 Fixed SSE reconnection scheduling and teardown edge cases after stop.
- 🐛 Fixed preview action consumer expectations in tests/UI to match the new typed `code` contract.
- 🔒 Added/strengthened cross-registry digest verification and release integrity checks; restricted cosign/container signing and provenance/SBOM attestation to prereleases; tightened release artifact/source SHA handling.
- ⚠️ Breaking: preview error/action consumers must migrate from `action.label` to `action.code` (`open-registry-settings` / `open-trigger-settings`).
- ⚠️ Breaking: internal registry/digest-cache lifecycle APIs now use return/accept “handle/generation” values (signature/behavior changes in lifecycle helpers).

## Concerns

- Validate all downstream integrations/tests are migrated from `action.label` to `action.code` (including any non-covered UI paths).
- Confirm GA-only `release-cut` dispatch inputs (`candidate_tag`, `candidate_digest`) and “exact seven-day-old prerelease RC candidate” logic match the intended release policy end-to-end.
- Ensure release finalization remains idempotent when re-dispatched mid-run (tag exists vs force-update; draft/public transitions; digest re-checks).
- Check for any remaining stale-preview race paths not using the new generation/invalidation guards.
- Review whether any memory/resource cleanup is needed for tracked SSE stream instances and digest-cache cycles on early exits/errors.
- Verify notification preview mock behavior continues to match production contracts (template override validation + rendered field expectations).

### CODE REVIEW

File: apps/demo/src/mocks/handlers/notifications.ts [hunk 1]
```diff
@@ -84,10 +92,43 @@ export const notificationHandlers = [
...
+    const rule = notificationRules.find((candidate) => candidate.id === params.id);
     if (!rule) return new HttpResponse(null, { status: 404 });
-    const body = (await request.json()) as Record<string, unknown>;
-    return HttpResponse.json({ ...rule, ...body });
+    const body = (await request.json()) as NotificationRuleUpdate;
+    Object.assign(rule, body, {
+      ...(body.triggers ? { triggers: [...body.triggers] } : {}),
+      ...(body.templates ? { templates: structuredClone(body.templates) } : {}),
+    });
+    return HttpResponse.json(rule);
```
[the remark this answers is not in the document]
Comment (scttbnsn) [reply apps/demo/src/mocks/handlers/notifications.ts:205]: Fixed in f379d3d9. All three demo handlers now share guarded JSON-object parsing, PATCH builds a validated allowlisted update, and contract tests cover malformed, null, and array bodies plus identity and arbitrary-field injection.

File: e2e/playwright/v16-modes-pins.spec.ts [hunk 2]
```diff
@@ -173,6 +173,7 @@ test.describe('v1.6 update modes, scheduling, and pinned tags', () => {
       container.displayName = 'Immich ML (Pinned)';
       container.tagFamily = 'strict';
       container.tagPinned = true;
+      container.tagPinGated = true;
```
[the remark this answers is not in the document]
Comment (scttbnsn) [reply e2e/playwright/v16-modes-pins.spec.ts:177]: Fixed in f379d3d9 by declaring tagPinGated as an optional boolean on ContainerFixture. The E2E Biome gate is green.

File: scripts/release-docs-identity.test.mjs [hunk 3]
```diff
@@ -336,16 +336,19 @@ test('current and archived docs describe destructive and recovery behavior accur
...
-        /Upgrade, no auth configured, no `DD_ANONYMOUS_AUTH_CONFIRM` \| Fails closed — all API calls return `401`/u,
+        /Upgrade, no auth configured, no `DD_ANONYMOUS_AUTH_CONFIRM` \| Starts fail-closed — protected API requests return `401`; auth discovery\/status stays public; `\/health` returns `503`/u,
+      );
+      assert.match(
+        authentications,
+        /The SPA shell may still load, but it cannot read protected application data/u,
       );
       assert.doesNotMatch(authentications, /refuses to start/u);
+      assert.doesNotMatch(authentications, /all API calls return `401`/u);
+      assert.doesNotMatch(authentications, /every API (call|request)/u);
```
[the remark this answers is not in the document]
Comment (scttbnsn) [reply scripts/release-docs-identity.test.mjs]: Fixed in f379d3d9. Both assertions now reuse one case-insensitive BROAD_401_CLAIM pattern covering all or every, call or request singular and plural, and return or rejected wording. The release-doc identity suite is green.

File: ui/tests/views/NotificationsView.spec.ts [hunk 4]
```diff
@@ -331,6 +331,68 @@ describe('NotificationsView', () => {
...
+
+    await wrapper.get('button[aria-label="Preview notification template"]').trigger('click');
+    const triggerSelect = wrapper.get('select[aria-label="Template trigger"]');
+    expect((triggerSelect.element as HTMLSelectElement).value).toBe('trigger:discord-alerts');
+    await triggerSelect.setValue('trigger:slack-alerts');
+    await nextTick();
+    expect((triggerSelect.element as HTMLSelectElement).value).toBe('trigger:slack-alerts');
+    const previewButton = wrapper.get('button[aria-label="Preview notification template"]');
+    expect(previewButton.attributes('disabled')).toBeUndefined();
+    await previewButton.trigger('click');
```
[the remark this answers is not in the document]
Comment (scttbnsn) [reply ui/tests/views/NotificationsView.spec.ts:374]: Verified as a false positive; no code change needed. The existing test asserts Discord is selected for the first request, then switches to Slack and verifies that the stale Discord result is discarded. The focused test passes unchanged.

File: scripts/release-docs-identity.test.mjs [hunk 5]
```diff
@@ -378,6 +381,42 @@ test('current and archived docs describe destructive and recovery behavior accur
...
+});
+
+test('security policy distinguishes stable and prerelease support', () => {
+  const security = read('SECURITY.md');
+
+  assert.match(security, /Latest stable release/u);
+  assert.match(security, /Latest release candidate on the active train/u);
+  assert.match(security, /Older release candidates are not patched/u);
+  assert.doesNotMatch(security, /\| latest\s+\|/u);
+});
```
[the remark this answers is not in the document]
Comment (scttbnsn) [reply scripts/release-docs-identity.test.mjs:415]: Fixed in f379d3d9 with a case-insensitive pattern and flexible whitespace around latest. The release-doc identity suite is green.
